Extremely long branch names created by Dependabot fail the build

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Package ecosystem

nuget

### Package manager version

_No response_

### Language version

_No response_

### Manifest location and content before the Dependabot update

_No response_

### dependabot.yml content

_No response_

### Updated dependency

_No response_

### What you expected to see, versus what you actually saw

Because Dependabot creates very long branch names, such as:
- dependabot-nuget-dot-config-dotnet-reportgenerator-globaltool-5.4.12
- dependabot-nuget-dot-config-jetbrains.resharper.globaltools-2025.2.0

Building the PR fails with errors on Windows/Linux/macOS, such as:

> Error: /Users/runner/.dotnet/sdk/9.0.304/Sdks/NuGet.Build.Tasks.Pack/buildCrossTargeting/NuGet.Build.Tasks.Pack.targets(221,5): error NU5123: Warning As Error: The file 'package/services/metadata/core-properties/27dfef446f6043e3a174e1942aa22383.psmdcp' path, name, or both are too long. Your package might not work without long file path support. Please shorten the file path or file name. [/Users/runner/work/JsonApiDotNetCore/JsonApiDotNetCore/src/JsonApiDotNetCore.OpenApi.Client.NSwag/JsonApiDotNetCore.OpenApi.Client.NSwag.csproj]
Error: /Users/runner/.dotnet/sdk/9.0.304/Sdks/NuGet.Build.Tasks.Pack/buildCrossTargeting/NuGet.Build.Tasks.Pack.targets(221,5): error NU5123: Warning As Error: The file 'package/services/metadata/core-properties/576f133caa684374a95b9df1e19e3221.psmdcp' path, name, or both are too long. Your package might not work without long file path support. Please shorten the file path or file name. [/Users/runner/work/JsonApiDotNetCore/JsonApiDotNetCore/src/JsonApiDotNetCore.OpenApi.Client.Kiota/JsonApiDotNetCore.OpenApi.Client.Kiota.csproj]

Examples of failing PR builds:
- https://github.com/json-api-dotnet/JsonApiDotNetCore/pull/1764
- https://github.com/json-api-dotnet/JsonApiDotNetCore/pull/1761

Please shorten the branch names created by Dependabot, so that builds succeed.

### Native package manager behavior

_No response_

### Images of the diff or a link to the PR, issue, or logs

_No response_

### Smallest manifest that reproduces the issue

_No response_

--- a/common/lib/dependabot/pull_request_creator.rb
+++ b/common/lib/dependabot/pull_request_creator.rb
@@ -197,7 +197,7 @@ module Dependabot
       milestone: nil,
       branch_name_separator: "/",
       branch_name_prefix: "dependabot",
-      branch_name_max_length: nil,
+      branch_name_max_length: 100,
       label_language: false,
       automerge_candidate: false,
       github_redirection_service: DEFAULT_GITHUB_REDIRECTION_SERVICE,

--- a/common/lib/dependabot/pull_request_creator/branch_namer/base.rb
+++ b/common/lib/dependabot/pull_request_creator/branch_namer/base.rb
@@ -1,6 +1,7 @@
 # typed: strong
 # frozen_string_literal: true
 
+require "digest"
 require "sorbet-runtime"
 
 module Dependabot
@@ -70,9 +71,10 @@ module Dependabot
           sanitized_name = sanitized_name.gsub("/", separator)
 
           # Shorten the ref in case users refs have length limits
-          if max_length && (sanitized_name.length > T.must(max_length))
-            sha = T.must(Digest::SHA1.hexdigest(sanitized_name)[0, T.must(max_length)])
-            sanitized_name[[T.must(max_length) - sha.size, 0].max..] = sha
+          branch_name_max_length = max_length
+          if branch_name_max_length && (sanitized_name.length > branch_name_max_length)
+            sha = T.must(Digest::SHA1.hexdigest(sanitized_name)[0, branch_name_max_length])
+            sanitized_name[[branch_name_max_length - sha.size, 0].max..] = sha
           end
 
           sanitized_name

--- a/common/spec/dependabot/pull_request_creator_spec.rb
+++ b/common/spec/dependabot/pull_request_creator_spec.rb
@@ -226,6 +226,63 @@ RSpec.describe Dependabot::PullRequestCreator do
       end
     end
 
+    context "with a dependency name that would generate a branch longer than 100 characters" do
+      let(:source) { Dependabot::Source.new(provider: "github", repo: "gc/bp", branch: "main") }
+      let(:dummy_creator) { instance_double(described_class::Github) }
+      let(:dependency) do
+        Dependabot::Dependency.new(
+          name: "this.is.an.extremely.long.package.name.that.will.definitely.exceed.the.maximum.length.limit",
+          version: "1.5.0",
+          previous_version: "1.4.0",
+          package_manager: "bundler",
+          requirements:
+            [{ file: "Gemfile", requirement: "~> 1.5.0", groups: [], source: nil }],
+          previous_requirements:
+            [{ file: "Gemfile", requirement: "~> 1.4.0", groups: [], source: nil }]
+        )
+      end
+
+      it "shortens the branch name to 100 characters by default" do
+        expect(described_class::Github)
+          .to receive(:new) do |**kwargs|
+            expect(kwargs[:branch_name].length).to eq(100)
+            dummy_creator
+          end
+        expect(dummy_creator).to receive(:create)
+        creator.create
+      end
+
+      context "when branch_name_max_length is explicitly set to nil" do
+        subject(:creator) do
+          described_class.new(
+            source: source,
+            base_commit: base_commit,
+            dependencies: dependencies,
+            files: files,
+            credentials: credentials,
+            custom_labels: custom_labels,
+            reviewers: reviewers,
+            assignees: assignees,
+            milestone: milestone,
+            author_details: author_details,
+            signature_key: signature_key,
+            provider_metadata: provider_metadata,
+            branch_name_max_length: nil
+          )
+        end
+
+        it "does not shorten the branch name" do
+          expect(described_class::Github)
+            .to receive(:new) do |**kwargs|
+              expect(kwargs[:branch_name].length).to be > 100
+              dummy_creator
+            end
+          expect(dummy_creator).to receive(:create)
+          creator.create
+        end
+      end
+    end
+
     context "with a GitLab source" do
       let(:source) { Dependabot::Source.new(provider: "gitlab", repo: "gc/bp", branch: "main") }
       let(:dummy_creator) { instance_double(described_class::Gitlab) }

